docs/data/joy/components/modal/DialogVerticalScroll.js

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import Button from '@mui/joy/Button';
+import List from '@mui/joy/List';
+import ListItem from '@mui/joy/ListItem';
+import FormControl from '@mui/joy/FormControl';
+import FormLabel from '@mui/joy/FormLabel';
+import Switch from '@mui/joy/Switch';
+import Modal from '@mui/joy/Modal';
+import ModalDialog from '@mui/joy/ModalDialog';
+import ModalClose from '@mui/joy/ModalClose';
+import Typography from '@mui/joy/Typography';
+import Stack from '@mui/joy/Stack';
+
+export default function DialogVerticalScroll() {
+  const [open, setOpen] = React.useState('');
+  const [scroll, setScroll] = React.useState(true);
+  return (
+    <React.Fragment>
+      <Stack direction="row" spacing={1}>
+        <Button variant="outlined" color="neutral" onClick={() => setOpen('center')}>
+          Center
+        </Button>
+        <Button
+          variant="outlined"
+          color="neutral"
+          onClick={() => setOpen('fullscreen')}
+        >
+          Full screen
+        </Button>
+      </Stack>
+      <Modal open={!!open} onClose={() => setOpen('')}>
+        <ModalDialog
+          aria-labelledby="dialog-vertical-scroll-title"
+          aria-describedby="dialog-vertical-scroll-description"
+          layout={open || undefined}
+        >
+          <ModalClose />
+          <Typography id="dialog-vertical-scroll-title" component="h2">
+            Vertical scroll example
+          </Typography>
+          <FormControl
+            orientation="horizontal"
+            sx={{ bgcolor: 'background.level2', p: 1, borderRadius: 'sm' }}
+          >
+            <FormLabel>Container overflow</FormLabel>
+            <Switch
+              checked={scroll}
+              onChange={(event) => setScroll(event.target.checked)}
+              sx={{ ml: 'auto' }}
+            />
+          </FormControl>
+          <List
+            sx={{
+              overflow: scroll ? 'scroll' : 'initial',
```

Comment:
I have almost never see a case where `overflow: scroll` makes sense. Here is another example that reinforces it. Here is how it looks on Windows:

<img width="454" alt="Screenshot 2023-03-05 at 20 07 45" src="https://user-images.githubusercontent.com/3165635/222980711-affcf8d9-4e25-434b-93ea-8b102d074082.png">

Same on macOS if you enable the nonempty scroll bar which I hight recommend as it matches Windows which is >50% of what end-users experience

<img width="379" alt="Screenshot 2023-03-05 at 20 10 44" src="https://user-images.githubusercontent.com/3165635/222980768-2b21ff2d-5e20-4a91-8d13-3bba5cbb782f.png">

auto will remove the scrollbar at the bottom.

```suggestion
              overflow: scroll ? 'auto' : 'initial',
```

Reply:
Another example why "nonempty scrollbar which I highly recommend", the transition jumps in on Windows

https://deploy-preview-36103--material-ui.netlify.app/joy-ui/react-modal/#transition

While it doesn't on [MUI Base](https://mui.com/base/react-modal/#transitions) and [Material UI](https://mui.com/material-ui/react-dialog/)